src/__tests__/wait-for-dom-change.js

```diff
@@ -48,3 +53,48 @@ Array [
 ]
 `)
 })
+
+test('always uses real timers', async () => {
+  const expectElementToChange = async () => {
```

Comment:
I think I'd prefer to put this function outside the test and have both cases as individual tests that call the function.